internal/resource/common/resourcelink/resource_link_client_struct.go

```diff
@@ -18,15 +18,10 @@ func ClientStruct(planObj basetypes.ObjectValue) (*client.ResourceLink, error) {
 	if !ok {
 		return nil, errors.New("object value missing \"id\" attribute when creating resource link client struct")
 	}
-	objLoc, ok := objValues["location"]
```

Comment:
We still want to set location on the other resources, right?

Reply:
will add boolean argument for this